dev/run-test-container.sh

```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
```

Comment:
Stop on failure. Otherwise, for example, if the image build fails, the container is still launched using an old image (confusing!).

Reply:
Can we move these changes in a separate PR?